### PR TITLE
itk4, vtk_7: remove

### PR DESCRIPTION
**Pull request by bcdarwin**
ants: 2.2.0 -> 2.4.1
itk-unstable: init at 2022-07-02

###### Description of changes

Updates ANTs, which requires a new ITK version (currently unstable, but ANTs should update to 5.3 shortly after it's released).  This allows us to drop old itk4 and vtk_7 from Nixpkgs.  (Note that even if we were to wait the ITK 5.3 release we would still need to maintain ITK 5.2.1 as well until all packages migrate to 5.3.)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mindavi — APPROVED** [review]
Looks like a good cleanup to me. Didn't test anything.

File: pkgs/development/libraries/itk/generic.nix [hunk 1]
```diff
@@ -22,8 +24,8 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "InsightSoftwareConsortium";
     repo = "ITK";
-    rev = "v${version}";
-    sha256 = "sha256-KaVe9FMGm4ZVMpwAT12fA67T0qZS3ZueiI8z85+xSwE=";
+    rev = rev;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/itk/generic.nix:27]:
```suggestion
    inherit rev;
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -18798,15 +18798,16 @@ with pkgs;
...
   };
 
-  itk4 = callPackage ../development/libraries/itk/4.x.nix {
-    stdenv = if stdenv.cc.isGNU && stdenv.system == "x86_64-linux" then gcc10Stdenv else stdenv;
+  itk_5 = callPackage ../development/libraries/itk/5.x.nix {
     inherit (darwin.apple_sdk.frameworks) Cocoa;
   };
 
-  itk = callPackage ../development/libraries/itk {
+  itk_5_3-unstable = callPackage ../development/libraries/itk/5.3-unstable.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:18805]:
```suggestion
  itk-unstable = callPackage ../development/libraries/itk/unstable.nix {
```

File: pkgs/development/libraries/itk/generic.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/libraries/itk/generic.nix]: Changed, thanks.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/top-level/all-packages.nix:18805]: Renamed.

**Review by tfmoraes — APPROVED** [review]

**Pull request merged**